kratos/sources/model_part_io.cpp

```diff
@@ -179,18 +179,37 @@ namespace Kratos
         std::string aux_string;
         const std::string string_to_remove = "This properties contains 0 tables";
 
-        for(PropertiesContainerType::const_iterator i_properties = rThisProperties.begin() ; i_properties != rThisProperties.end() ; i_properties++)
-        {
+        for (const auto i_properties = rThisProperties.begin() ; i_properties != rThisProperties.end() ; ++i_properties) {
             std::ostringstream aux_ostream;
             (*mpStream) << "Begin Properties " << i_properties->Id() << std::endl;
             i_properties->PrintData(aux_ostream);
 
             aux_string = aux_ostream.str();
-            std::string::size_type i = aux_string.find(string_to_remove);
 
-            if (i != std::string::npos)
-            {
-                aux_string.erase(i, string_to_remove.length());
+            // We remove the line of Constitutive Laws and we add it manually
+            if (i_properties->Has(CONSTITUTIVE_LAW)) {
+                std::string::size_type it_constitutive_law_begin = aux_string.find("CONSTITUTIVE_LAW");
+                std::string::size_type it_constitutive_law_end = aux_string.find("\n", it_constitutive_law_begin);
+
+                if (it_constitutive_law_begin != std::string::npos) {
+                    aux_string.erase(it_constitutive_law_begin, it_constitutive_law_end);
+                }
+                const ConstitutiveLaw::Pointer p_law = i_properties->GetValue(CONSTITUTIVE_LAW);
+                auto components_cl = KratosComponents<ConstitutiveLaw>::GetComponents();
+                std::string cl_name = "";
+                for (const auto& comp_cl : components_cl) {
+                    if (p_law->HasSameType(p_law.get(), comp_cl.second)) {
```

Comment:
what are u trying to do here?

Reply:
Now when you use the model part io, instead of the name of the cl it returns you the pointer (literally the hash of the pointer). This is the same it is done in the elements. You need to do a comparison with typo info

Reply:
Described now